modules/null/agent-gw-cluster-setup/setup.tftpl

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+#set -x
+
+cookie_file=$(mktemp)
+
+trap "curl -k -s --cookie $cookie_file --request DELETE https://${mx_address}:${mx_port}/SecureSphere/api/v1/auth/session; rm -f $cookie_file" EXIT
+
+function moveGatewayToCluster() {
+  gw_name=$1
+  response=$(curl -k -s -X PUT -b $cookie_file "https://${mx_address}:${mx_port}/SecureSphere/api/v1/conf/gateways/$gw_name/move/${cluster_name}")
+
+  if echo $response | grep 'errors' &>/dev/null; then
+    echo "Failed move gateway $gw_name to cluster ${cluster_name}. MX response: $response"
+    exit 1
+  else
+    echo "Gateway $gw_name was moved to cluster ${cluster_name}"
+  fi
+}
+
+response=$(curl -k -s -X POST --cookie-jar $cookie_file "https://${mx_address}:${mx_port}/SecureSphere/api/v1/auth/session" \
+  --header "Authorization: Basic ${https_auth_header}")
+
+if echo $response | grep 'errors' &>/dev/null; then
+  echo "Failed to login to the MX. MX response: $response"
+  exit 1
+fi
+
+# Step 2: Get all gateways in the gateway group
+response=$(curl -k -s -X GET -b $cookie_file "https://${mx_address}:${mx_port}/SecureSphere/api/v1/conf/gatewayGroups/${gateway_group_name}/gateways")
+
+if echo $response | grep 'errors' &>/dev/null; then
+  echo "Failed get all gateways in the gateway group. MX response: $response"
+  exit 1
+fi
+
+gw_names=($(echo "$response" | jq -r '.gateways[].name'))
+
+if [ -z "$gw_names" ]; then
+  echo "Failed to extract the gateway names in the gateway group"
+  exit 1
+fi
+
+gw_count=$${#gw_names[@]}
+
+if [ "$gw_count" -ge 2 ]; then
+  echo "The number of gateways in '${gateway_group_name}' is $gw_count"
+else
+  echo "Expected at least 2 gateways in gateway group '${gateway_group_name}', which is the minimum number required to setup a Cluster, but was $gw_count"
+  exit 1
+fi
+
+# Step 3: Create an empty cluster
+request_body="{\"name\":\"${cluster_name}\", \"clusterType\":\"REGULAR_CLUSTER\", \"autoLoadBalance\":true, \"hasRedundant\":true, \"networkTopology\":\"Single\"}"
```

Comment:
It is highly possible, it will come at a later stage. In addition, I assume you have the same comment for the update of the cluster, in that case, I think there even might be some problems with the arguments I send which are causing my cluster to be in "split brain" state or to have some other failure, it is also possible that I drove the GWs mad with the 10s of move/activate/disable etc. operations which I did. After I update the example, I will run a fresh deployment and look into these issues.